src/app/shared/plunker/plunker-writer.ts

```diff
@@ -113,9 +113,25 @@ export class PlunkerWriter {
       fileContent = fileContent.replace(/material-docs-example/g, data.selectorName);
     } else if (fileName == 'main.ts') {
       // Replace the component name in `main.ts`.
-      // For example, `import {MaterialDocsExample} from 'material-docs-example'`
+
+      // Replace `import {MaterialDocsExample} from 'material-docs-example'`
       // will be replaced as `import {ButtonDemo} from './button-demo'`
-      fileContent = fileContent.replace(/MaterialDocsExample/g, data.componentName);
+      fileContent = fileContent.replace(/{MaterialDocsExample}/g, `{${data.componentName}}`);
+
+      // Replace `declarations: [MaterialDocsExample]`
+      // will be replaced as `declarations: [ButtonDemo]`
+      fileContent = fileContent.
+        replace(/declarations: \[MaterialDocsExample\]/g,
```

Comment:
Also replace `entryComponents`? 

Add `entryComponents: [MaterialDocsExample]` to `main.ts`.

I tested on plunker and found out it doesn't work without entryComponents.